### PR TITLE
(feat) Migrating allergy form to RHF and Zod 

### PR DESCRIPTION
**Pull request by lumuchris256**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Migrated the form to use reacthookform and zod 

## Screenshots
https://www.loom.com/share/b93b8fcb2a37447697532163f6c4ce02?sid=81bddfd3-c160-4f88-bba4-3db16e5b5ab1

## Related Issue
https://issues.openmrs.org/browse/O3-2143

## Other

cc @denniskigen @hadijahkyampeire

### CODE REVIEW

**Review by hadijahkyampeire — CHANGES_REQUESTED** [review]
@lumuchris256 I realized in your demo that the save and close button is always enabled but doesn't even save until all the filled are chosen, could you please disable it instead or add warning validations that tell the user that they need to complete the whole form.

File: packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.test.tsx [hunk 1]
```diff
@@ -78,7 +78,6 @@ describe('AllergyForm ', () => {
     expect(screen.getByRole('textbox', { name: /date of onset and comments/i })).toBeInTheDocument();
     expect(screen.getByRole('button', { name: /discard/i })).toBeInTheDocument();
     expect(screen.getByRole('button', { name: /save and close/i })).toBeInTheDocument();
-    expect(screen.getByRole('button', { name: /save and close/i })).toBeDisabled();
```
Comment (hadijahkyampeire) [inline packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.test.tsx:81]: If you fix what I mentioned above, then this test won't need to go away.

**Comment by lumuchris256** [comment]
reguested change
[Screencast from 2023-08-27 12-23-51.webm](https://github.com/openmrs/openmrs-esm-patient-chart/assets/59338693/03c824fc-195b-4f5a-a1c2-b33b105359cc)

**Review by hadijahkyampeire — DISMISSED** [review]
Thanks @lumuchris256

**Comment by hadijahkyampeire** [comment]
@lumuchris256 please fix the e2e tests, I see they are related to allergies.

**Comment by lumuchris256** [comment]
> @lumuchris256 please fix the e2e tests, I see they are related to allergies.

@hadijahkyampeire  the form is ready for your re-review

**Review by denniskigen — APPROVED** [review]
Thanks, @lumuchris256!

**Pull request merged**